pkgs/applications/editors/jetbrains/update.py

```diff
@@ -63,18 +62,21 @@ def update_product(name, product):
     else:
         try:
             build = latest_build(channel)
-            version = build["@version"]
-            parsed_version = build_version(build)
-            version_major_minor = f"{parsed_version.major}.{parsed_version.minor}"
-            download_url = product["url-template"].format(version = version, versionMajorMinor = version_major_minor)
+            if "EAP" not in channel["@name"]:
+                version_or_build_number = build["@version"]
+            else:
+                version_or_build_number = build["@fullNumber"]
+            version_number = build["@version"].split(' ')[0]
+            download_url = product["url-template"].format(version=version_or_build_number, versionMajorMinor=version_number)
+            product["version-major-minor"] = version_number
             product["url"] = download_url
-            product["version-major-minor"] = version_major_minor
-            if "sha256" not in product or product.get("version") != version:
-                logging.info("Found a newer version %s.", version)
-                product["version"] = version
+            product["version"] = build["@version"]
```

Comment:
Seems this line is not needed at all. If `version` requires update the we will get into the `if` statement and update it, otherwise it should not be changed.

Reply:
Fixed it.